mtwitter.js

```diff
@@ -0,0 +1,291 @@
+var Twitter = require('twitter');
+var Request = require('request');
+var fs = require('fs');
+var Buffer = require('buffer').Buffer;
+
+const TWITTER_PROFILE_INTERVAL = 60000;//ms;
+
+var MTwitter = function (bridge, config) {
+  this.app_auth = config.app_auth;
+  this.app_twitter = null;
+  this.tuser_cache = {};
+  this.timeline_list = [];
+  this.timeline_queue = [];
+  this.timeline_period = 0;
+  this.timeline_intervalID = null;
+  this.msg_queue = [];
+  this._bridge = bridge;
+
+
+  this.get_bearer_token().then((token) => {
+    this.app_auth.bearer_token = token;
+    console.log('Retrieved token');
+    this.app_twitter = new Twitter(this.app_auth);
+    setInterval(() => { this._process_queue(); }, 500);
+    this.start_timeline();
+  }).catch((error) => {
+      console.error('Error trying to retrieve bearer token:', error);
+      throw "Couldn't get a bearer token for Twitter AS.";
+  });
+};
+
+MTwitter.prototype._get_bearer_http = function () {
+  return new Promise( (resolve,reject) => {
+    var key = this.app_auth.consumer_key + ":" + this.app_auth.consumer_secret;
+    key = Buffer.from(key, 'ascii').toString('base64');
+    var options = {
+        url: "https://api.twitter.com/oauth2/token",
+        headers: {
+            'Authorization': "Basic " + key
+        },
+        form: "grant_type=client_credentials",
+        contentType: "application/x-www-form-urlencoded;charset=UTF-8"
+    };
+    Request.post(options, function(error, response, body) {
+        if (error) {
+            reject(error);
+        } else {
+            try {
+                var jsonresponse = JSON.parse(body);
+                if (jsonresponse.token_type == "bearer") {
+                    fs.writeFile("bearer.tok", jsonresponse.access_token, (err) => {
+                        if (err) {
+                            //This error is unfortunate, but not a failure to retrieve a token so the bridge can run fine.
+                            console.error("Couldn't write bearer token to file. Reason:",err);
+                        }
+                    });
+                    //Not waiting for callback since it is trivial to get a new token, and can be done async
+                    resolve(jsonresponse.bearer_token);
+                } else {
+                    reject("Request to oauth2/post did not return the correct token type ('bearer'). This is weeeird.");
```

Comment:
Add in the `body` to this error message so in the unlikely event this *does* fire, we can immediately see what Twitter returned instead.